crates/oxc_isolated_declarations/src/declaration.rs

```diff
@@ -134,7 +134,12 @@ impl<'a> IsolatedDeclarations<'a> {
                     decl.span,
                     decl.id.clone_in(self.ast.allocator),
                     Some(TSModuleDeclarationBody::TSModuleDeclaration(inner)),
-                    decl.kind,
+                    // Follows https://github.com/microsoft/TypeScript/pull/54134
+                    if decl.kind == TSModuleDeclarationKind::Module {
+                        TSModuleDeclarationKind::Namespace
+                    } else {
+                        decl.kind
+                    },
```

Comment:
I think we can use `TSModuleDeclarationKind::Namespace` directly here, as no `global {}` would enter this method.